tomcat/src/main/java/org/apache/coyote/http11/message/request/HttpRequestReader.java

```diff
@@ -0,0 +1,64 @@
+package org.apache.coyote.http11.message.request;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.coyote.http11.message.HttpHeaders;
+
+// TODO: BufferedReader와 BufferedInputStream 같이 사용했을 때 문제가 발생하지 않는지 확인하기
+public class HttpRequestReader {
+
+    private final InputStream inputStream;
+
+    public HttpRequestReader(InputStream inputStream) {
+        this.inputStream = inputStream;
+    }
+
+    public HttpRequest read() throws IOException {
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+        String requestLine = bufferedReader.readLine();
+        HttpHeaders headers = readHeaderLines(bufferedReader);
+
+        return HttpRequest.of(requestLine, headers, readBody(headers));
+    }
+
+    private HttpHeaders readHeaderLines(BufferedReader bufferedReader) throws IOException {
+        List<String> headerLines = new ArrayList<>();
+
+        String headerLine = bufferedReader.readLine();
+        while (!headerLine.isEmpty()) {
```

Comment:
```suggestion
    while ((headerLine = bufferedReader.readLine()) != null && !headerLine.isEmpty()) {
```
null 검사도 추가하면, 예상치 못한 상황에 대해서도 안전하게 파싱할 수 있을 것 같습니다ㅎㅎ

Reply:
훨씬 안전하겠네요!!
해당 부분 리팩토링 하겠습니다!

꼼꼼한 리뷰 감사드립니다 ㅎㅎ